tests/test_toc_llm_analyser.py

```
@@ -0,0 +1,35 @@
+import unittest
+from typing import cast
+
+from pdf_craft.llm.types import Message, MessageRole
+from pdf_craft.llm.core import LLM
+from pdf_craft.toc.llm_analyser import LLMAnalysisError, _LLMAnalyser
+
+
+class _BrokenLLM:
+    def request(self, input):  # pylint: disable=redefined-builtin,unused-argument
```

Ruff (0.15.9)

[error] Function argument `input` is shadowing a Python builtin

(A002)